sound/soc/sof/intel/hda.c

```diff
@@ -413,6 +417,10 @@ static int hda_init_caps(struct snd_sof_dev *sdev)
 
 	device_disable_async_suspend(&pci->dev);
 
+	/* enable ppcap interrupt */
+	hda_dsp_ctrl_ppcap_enable(sdev, true);
+	hda_dsp_ctrl_ppcap_int_enable(sdev, true);
+
```

Comment:
why do we need to do things differently? You could just use the second version everywhere?

Reply:
> why do we need to do things differently? You could just use the second version everywhere?

we have 3 ways to do the same thing:
1. using hda lib
2. we wrapped a API ourselves such as hda_dsp_ctrl_ppcap_xxx
3. directly write the registers 
I heard from @keyonjie  that it is because the license of GPL/BSD so we keep using hda lib when **IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)** is **true**.
And it seems we need to regulate where to use the 3 ways.
At least if we had already defined a API for a register read/write sequences, use the API instead of use the same direct registers read/write again will make the code more clear.

I made here differently is because that the sequence here are highly coherent with the hda_dsp_resume. if we can wrap it with a function, then in hda_dsp_resume we can directly call the function.

Reply:
You misunderstood the direction:
if SND_SOC_SOF_HDA is not defined, we shall not use any code from the hdac library. We may use structures from include files (as done for streams).

If SND_SOC_SOF_HDA is defined, we *may* use code from the hdac library, but if we already have code in SOF we should be consistent and use this only. In other words, the SOF code is trying to be self-contained and use the hdac library only if strictly necessary. Since we already have a replacement for this ppcap handling, let's use it across the board in all cases.
Makes sense?

Reply:
@plbossart OK, thanks! will update the PR.